File: next/types/CourseSubscription.ts
import { CourseProgress } from "@/courses/courses-hub/types/progress/CourseProgress";

export interface ICourseSubscription {
    id: string;
    subId: string;
    userId: string;
    isFavourite?: boolean;
    isSubscribed?: boolean;

    roles?: string[];

    progress?: CourseProgress;
    courseVersion?: string;
}

File: kotlin/coretypings/src/main/kotlin/eduflow/course/subscription/CourseSubscription.kt
package eduflow.course.subscription

import eduflow.course.subscription.progress.CourseProgress
import java.util.*

interface CourseSubscription {
    val _id: String
    val userId: String
    val courseId: String

    val isFavourite: Boolean?
    val isSubscribed: Boolean?
    val roles: List<String>?
    val createdAt: Date
    val updatedAt: Date

    val courseVersion: String?
    val progress: CourseProgress?
}